Bugfix: cgroups configuration

ppenchev-storpool opened the pull request:
Do not install the cgroups configuration at all when running in a container (e.g. the cinder-storpool charm).
Do not install (and remove) a stale configuration file.

ppenchev-storpool committed:
Allow the non-StorPool cgroups to swap.

ppenchev-storpool committed:
Do not install the stale storpool_cgmove.cron file.

ppenchev-storpool committed:
Do not configure cgroups in a container.

ryan-beisner commented:
Unit tests on this branch are failing:

```
pep8 installed: charmhelpers==0.18.5,charms.reactive==0.6.2,cliff==2.11.0,cmd2==0.8.1,extras==1.0.0,fixtures==3.0.0,flake8==3.5.0,future==0.16.0,Jinja2==2.10,linecache2==1.0.0,MarkupSafe==1.0,mccabe==0.6.1,mock==2.0.0,netaddr==0.7.19,os-testr==1.0.0,pbr==3.1.1,pkg-resources==0.0.0,prettytable==0.7.2,pyaml==17.12.1,pycodestyle==2.3.1,pyflakes==1.6.0,pyparsing==2.2.0,pyperclip==1.6.0,python-mimeparse==1.6.0,python-subunit==1.2.0,PyYAML==3.12,six==1.11.0,stestr==2.0.0,stevedore==1.28.0,Tempita==0.5.2,testtools==2.3.0,traceback2==1.4.0,unittest2==1.1.0,voluptuous==0.11.1
pep8 runtests: PYTHONHASHSEED='797746323'
pep8 runtests: commands[0] | flake8 reactive
pep8 runtests: commands[1] | flake8 --ignore=E402 unit_tests
unit_tests_35 installed: charmhelpers==0.18.5,charms.reactive==0.6.2,cliff==2.11.0,cmd2==0.8.1,extras==1.0.0,fixtures==3.0.0,flake8==3.5.0,future==0.16.0,Jinja2==2.10,linecache2==1.0.0,MarkupSafe==1.0,mccabe==0.6.1,mock==2.0.0,netaddr==0.7.19,os-testr==1.0.0,pbr==3.1.1,pkg-resources==0.0.0,prettytable==0.7.2,pyaml==17.12.1,pycodestyle==2.3.1,pyflakes==1.6.0,pyparsing==2.2.0,pyperclip==1.6.0,python-mimeparse==1.6.0,python-subunit==1.2.0,PyYAML==3.12,six==1.11.0,stestr==2.0.0,stevedore==1.28.0,Tempita==0.5.2,testtools==2.3.0,traceback2==1.4.0,unittest2==1.1.0,voluptuous==0.11.1
unit_tests_35 runtests: PYTHONHASHSEED='797746323'
unit_tests_35 runtests: commands[0] | ostestr
{0} unit_tests.test_common.TestStorPoolCommon.test_copy_config_files [0.004913s] ... ok
{0} unit_tests.test_common.TestStorPoolCommon.test_install_package [0.040874s] ... FAILED

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    b'Traceback (most recent call last):'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/unit_tests/test_common.py", line 114, in inner1'
    b'    return inner2()'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/.tox/unit_tests_35/lib/python3.5/site-packages/mock/mock.py", line 1305, in patched'
    b'    return func(*args, **keywargs)'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/unit_tests/test_common.py", line 112, in inner2'
    b'    return f(inst, *args, **kwargs)'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/.tox/unit_tests_35/lib/python3.5/site-packages/mock/mock.py", line 1305, in patched'
    b'    return func(*args, **keywargs)'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/unit_tests/test_common.py", line 273, in test_install_package'
    b'    self.assertEquals(count_log + 6, h_log.call_count)'
    b'  File "/usr/lib/python3.5/unittest/case.py", line 1311, in deprecated_func'
    b'    return original_func(*args, **kwargs)'
    b'  File "/usr/lib/python3.5/unittest/case.py", line 820, in assertEqual'
    b'    assertion_func(first, second, msg=msg)'
    b'  File "/usr/lib/python3.5/unittest/case.py", line 813, in _baseAssertEqual'
    b'    raise self.failureException(msg)'
    b'AssertionError: 6 != 4'
    b''

==============================
Failed 1 tests - output below:
==============================

unit_tests.test_common.TestStorPoolCommon.test_install_package
--------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    b'Traceback (most recent call last):'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/unit_tests/test_common.py", line 114, in inner1'
    b'    return inner2()'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/.tox/unit_tests_35/lib/python3.5/site-packages/mock/mock.py", line 1305, in patched'
    b'    return func(*args, **keywargs)'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/unit_tests/test_common.py", line 112, in inner2'
    b'    return f(inst, *args, **kwargs)'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/.tox/unit_tests_35/lib/python3.5/site-packages/mock/mock.py", line 1305, in patched'
    b'    return func(*args, **keywargs)'
    b'  File "/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/unit_tests/test_common.py", line 273, in test_install_package'
    b'    self.assertEquals(count_log + 6, h_log.call_count)'
    b'  File "/usr/lib/python3.5/unittest/case.py", line 1311, in deprecated_func'
    b'    return original_func(*args, **kwargs)'
    b'  File "/usr/lib/python3.5/unittest/case.py", line 820, in assertEqual'
    b'    assertion_func(first, second, msg=msg)'
    b'  File "/usr/lib/python3.5/unittest/case.py", line 813, in _baseAssertEqual'
    b'    raise self.failureException(msg)'
    b'AssertionError: 6 != 4'
    b''


======
Totals
======
Ran: 2 tests in 0.6951 sec.
 - Passed: 1
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 1
Sum of execute time for each test: 0.0458 sec.

==============
Worker Balance
==============
 - Worker 0 (2 tests) => 0:00:00.046951

Slowest Tests:
Test id                                                           Runtime (s)
----------------------------------------------------------------  -----------
unit_tests.test_common.TestStorPoolCommon.test_install_package    0.041
unit_tests.test_common.TestStorPoolCommon.test_copy_config_files  0.005
ERROR: InvocationError: '/home/ubuntu/git/storpool-charms/storpool-charms/layers/layer-storpool-common/.tox/unit_tests_35/bin/ostestr'
___________________________________ summary ____________________________________
  pep8: commands succeeded
ERROR:   unit_tests_35: commands failed
```

thedac reviewed (changes requested):
The charms should not configure cgroups on physical hosts.

See https://github.com/storpool/layer-storpool-common/issues/3

ppenchev-storpool committed:
Merge the testing fixes into the fix-cgroups branch.

ppenchev-storpool committed:
Test more stages, but not install_package().

The install_package() and configure_cgroups() functions are not
very easy to test outside of a controlled environment.  Instead,
break some smaller building blocks out and test them.

ppenchev-storpool committed:
Install the latest StorPool version available.

Ignore the storpool_version charm configuration variable; StorPool will
take care of keeping compatible versions in our PPA.

This allows automated upgrades with no operator involvement.

ppenchev-storpool committed:
repo: upgrade the installed StorPool packages.

If we try to install the common packages with a new version while
there are, say, beacon and block packages at the old version,
the simple apt-get install for the common packages will fail, since
dependencies will be broken.  So look for StorPool packages and
upgrade them all.

ajkavanagh commented:
As @thedac points out, we still have a brittleness factor with configuring cgroups on the hosts; LXD and/or Juju may break this configuration leading to hard-to-debug issues.  A different way needs to be found.
